apps/accelerate/chatllama/chatllama/rlhf/model_loader.py

```diff
@@ -0,0 +1,258 @@
+
+        Args:
+            config (ConfigType): the config object, contains info of the model
+            is_checkpoint (bool): if True, the path is for a checkpoint
+            current_epoch (Optional[int]): the current epoch, used to create
+                the checkpoint name. If is_checkpoint is True, and
+                current_epoch is None, return just the folder and the simple
+                model name for the possible checkpoint.
+
+        Returns:
+            model_folder (str): the folder where the model is saved
+            model_name (str): the name of the model
+            path (Optional[str]): the path to the model. If is_checkpoint is
+                True, and current_epoch is None, return None
+        """
+        # Get model folder from settings (i.e.  base path for all the models)
+        if isinstance(config, ConfigActor) or isinstance(config, ConfigReward):
+            model_folder = config.model_folder
+        elif isinstance(config, Config):
+            model_folder = config.actor.model_folder
+        else:
+            raise ValueError(
+                "Config type not recognized during saving or loading"
+            )
+
+        # Add the checkpoint path if necessary
+        if is_checkpoint:
+            model_folder = os.path.join(model_folder, "checkpoints")
+
+        # Create the folder for the model type
+        #  (Actor, Critic, Reward, Actor_RL)
+        if isinstance(config, ConfigReward):
+            # here use ad-hoc flag from config to distinguish between
+            #  reward and critic
+            if config.is_reward:
+                model_folder = os.path.join(model_folder, "reward")
+            else:
+                model_folder = os.path.join(model_folder, "critic")
+        elif isinstance(config, ConfigActor):
+            model_folder = os.path.join(model_folder, "actor")
+        elif isinstance(config, Config):
+            model_folder = os.path.join(model_folder, "actor_rl")
+
+        # Make the path if not exists
+        if os.path.exists(model_folder) is False:
+            os.makedirs(model_folder)
+            print(f"Model folder does not exist. Creating it: {model_folder}")
+
+        # Create the model name
+        model_name = None
+        if isinstance(config, Config):
+            model_name = config.actor.model
+        elif isinstance(config, ConfigReward) or isinstance(
+            config, ConfigActor
+        ):
+            model_name = config.model
+        if model_name in hf_models:
+            model_name = os.path.split(model_name)[-1]
+        if model_name is None:
+            raise ValueError("Model name not found")
+
+        # If is a checkpoint and current epoch are available
+        # extend the model name with the epoch, if none epoch is provided
+        # just return the simple model name
+        if is_checkpoint and current_epoch is not None:
+            # number of characters to store the checkpoints
+            n_char = 8
```

Comment:
This number should probably be estimated from the `max_epochs` or `total_steps` parameter.
Something like 
```python
n_char = max(len(str(max_epochs)), len(str(total_steps)))
```